### DISCUSSION TITLE
world_to_viewport works incorrectly? and a few other questions about the coordinate system

### DISCUSSION THREAD

**Discussion by Gor-Madatyan** [bevyengine/bevy#9210] [Q&A] [answered]
# Questions

## Which coordinate systems does  Bevy use? and what are their descriptions

As far as I know, those are `world`,`viewport`,`local(for children of some entity)` coordinate systems.

And are the local coordinates applicable only to the children of the camera? Or Not?

## how to get world | viewport coordinates from local coordinates?

# :1234: 

### (tested in 0.10, 0.11)

For some reason, world_to_viewport, viewport_to_world_2d and others calculates coordinates relative to top-right corner of the screen world_to_viewport points to the upper right corner when converting [0.0, 0.0], but was supposed to show the upper left corner according to [`this`](https://bevyengine.org/learn/migration-guides/0.10-0.11/#consistent-screen-space-coordinates)

some info about the system:
```

AdapterInfo { name: "GeForce RTX 2060", vendor: 4318, device: 7957, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "451.67", backend: Vulkan }
SystemInfo { os: "Windows 11 Home", kernel: "22621", cpu: "Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz", core_count: "6", memory: "15.8 GiB" }

```

Code which confirms my words

```rust
use bevy::prelude::*;

fn main() {
    App::new().add_plugins(DefaultPlugins).add_systems(Update,get_space_center).add_systems(Startup, create_camera).run();
}


fn create_camera(mut commands: Commands) {
    commands.spawn(Camera2dBundle::default());
}


fn get_space_center(mut commands:Commands,asset_server:Res<AssetServer>, query: Query<(&Camera,&GlobalTransform)>) {
    let camera = query.get_single().unwrap();
    let cords=camera.0.world_to_viewport(camera.1, Vec3::new(0.0,0.0,0.0)).unwrap();
commands.spawn(SpriteBundle {
        texture: asset_server.load("planet00.png"),
        transform:Transform::from_translation(Vec3::new(cords.x,cords.y,0.0)),
        ..Default::default()
    });
    dbg!(cords);
}

```

![Screenshot 2023-07-19 204356](https://github.com/bevyengine/bevy/assets/126885425/3d0a1587-d771-4dbe-b242-d862e438d560)

**Comment by superdump** [comment]
Right-handed, x-right, y-up, z-back and cameras look along -z.

**Comment by tim-blackbird** [comment, marked as answer]
Methods like `world_to_viewport` convert between 'viewport space' and 'world space'; they don't convert a world space position to another world space position that is relative to the viewport.
`world_to_viewport` is useful when you want to position a UI element (which are positioned in 'viewport space') relative to something in world space (like a sprite).

If you want to place an image relative to the window you probably want to take a look at the [ui examples](https://github.com/bevyengine/bevy/tree/latest/examples/ui) or [`ImageBundle`](https://docs.rs/bevy/latest/bevy/prelude/struct.ImageBundle.html) specifically.

**Reply by Gor-Madatyan** [reply]
Thank you, it explains all
